packages/mobile/src/identity/contactMapping.ts

```diff
@@ -218,19 +221,44 @@ export function* fetchAddressesAndValidateSaga({
   }
 }
 
-function* getAddresses(e164Number: string) {
-  let phoneHash: string
-  if (features.USE_PHONE_NUMBER_PRIVACY) {
-    const phoneHashDetails: PhoneNumberHashDetails = yield call(fetchPhoneHashPrivate, e164Number)
-    phoneHash = phoneHashDetails.phoneHash
-  } else {
-    phoneHash = getPhoneHash(e164Number)
-  }
-
+function* getAccountAddresses(e164Number: string) {
+  const phoneHashDetails: PhoneNumberHashDetails = yield call(fetchPhoneHashPrivate, e164Number)
+  const phoneHash = phoneHashDetails.phoneHash
   const lookupResult: IdentifierLookupResult = yield call(lookupAttestationIdentifiers, [phoneHash])
   return getAddressesFromLookupResult(lookupResult, phoneHash) || []
 }
 
+function* getWalletAddressesAndUpdateCache(e164Number: string) {
```

Comment:
Are we caching the account address as well? If not, how will we look up e.g. the DEK?

IMO the best thing would probably be to have an `Account` object, which consists of an account address, DEK, wallet address, verification info, etc., and have a cached mapping for which phone numbers are the keys and `Account`s are the values (well, probably actually two mappings, one from phone number -> (pepper, identifier), and one from identifier -> Account). And various operations on the `Account` object that allow us to refresh some (or all) of the cached values.

Reply:
Yeah see the `updateWalletToAccountAddress` action below. We are storing a mapping of `walletAddress => accountAddress` specifically for DEK lookups.

I agree that there is probably a better way to organize and consolidate the disparate caches within Valora but I am inclined to kick that can down the road for now.

Reply:
Agree it doesn't make sense to do now, just flagging for the future since it looks like this will be worth cleaning up a bit when we have time